botocore/credentials.py

```diff
@@ -406,6 +400,22 @@ def create_from_metadata(cls, metadata, refresh_using, method):
         )
         return instance
 
+    @property
+    def _advisory_refresh_timeout(self):
```

Comment:
Not sure if this is an acceptable way to handle this but just offering it as a potential suggestion.

Reply:
Might not also want to pull from the environment variables every time this property is called.

Reply:
Why not just save it off in the constructor rather than making a property method?